pkg/database/database.go

```diff
@@ -42,10 +49,9 @@ func NewDatabase(ctx context.Context, log *logrus.Entry, env env.Interface, m me
 		},
 	}
 
-	err = api.AddExtensions(&h.BasicHandle)
-	if err != nil {
-		return nil, err
-	}
+	h.SetInterfaceExt(reflect.TypeOf(api.SecureByte{}), 1, SecureByteExt{Cipher: cipher, Log: log})
+	h.SetInterfaceExt(reflect.TypeOf((*api.SecureString)(nil)), 1, SecureStringExt{Cipher: cipher, Log: log})
+	h.RawBytesExt = NoopExt{}
```

Comment:
wouldn't this break serialising new `[]byte` fields that we might add to the API?  I think your `ConvertExt` functions need to return a string.

Reply:
fixed. This came to me too late in the evening too :/